engine/src/main/java/com/google/android/fhir/FhirEngineProvider.kt

```diff
@@ -79,6 +86,8 @@ object FhirEngineProvider {
     return checkNotNull(fhirServices)
   }
 
+  @PublishedApi @Synchronized internal fun getFhirDataStore() = fhirDataStore
```

Comment:
By doing following you can eliminate storing context in FhirEngineConfiguration
1. Pass context in this API 
2. Store FhirDataStore in FhirServices


```suggestion
  @PublishedApi @Synchronized internal fun getFhirDataStore(context: Context): FhirDataStore {
    return getOrCreateFhirService(context).fhirDataStore
  }
```

Reply:
done.